commons/ava_networks/high_level_gecko_client.go

```diff
@@ -27,15 +27,18 @@ const (
 type HighLevelGeckoClient struct {
 	client    *gecko_client.GeckoClient
 	geckoUser *GeckoUser
+	networkAcceptanceTimeoutInSeconds int
 }
 
 func NewHighLevelGeckoClient(
 		client *gecko_client.GeckoClient,
 		username string,
-		password string) *HighLevelGeckoClient {
+		password string,
+		networkAcceptanceTimeoutInSeconds int) *HighLevelGeckoClient {
```

Comment:
Does it make sense to have the same timeout for everything?? I'd been imagining that there'd be different acceptance times for different things so that we'd actually want the timeouts to the lowest-level functions (e.g. `waitForXchainTransactionAcceptance`) to be function arguments, and then the "wrapper" functions that combine these to have an argument per timeout that it uses to make explicit to the user what timeouts it will be waiting for.

If all operations that this HighLevelGeckoClient does will take the same time to be accepted by the network then maybe it makes sense to do here, but my gut is guessing that the timeout will probably depend on what you're trying to do (?)

Reply:
so they are all expected to take roughly the same time - they're all essentially waiting for the network to accept a transaction and then that transaction acceptance to be represented somehow in the internal representation of the chain on the node being queried

Reply:
i'm concerned about the space of configurability with making every single one independently configurable, you end up increasing the number of points that you have to modify in order to adjust for different environments which can be a real drag to deployment as they're throughout the code base